spec/integration/mentioning_spec.rb

```diff
@@ -22,74 +35,275 @@ def mention_stream_for(user)
   def users_connected?(user1, user2)
     user1.contacts.where(person_id: user2.person).count > 0
   end
-end
 
+  def post_status_message(user, aspects)
+    status_msg = nil
+    inlined_jobs do
+      post "/status_messages.json", status_message: {text: text_mentioning(user)}, aspect_ids: aspects
+      status_msg = StatusMessage.find(JSON.parse(response.body)["id"])
+    end
+    status_msg
+  end
+
+  def receive_each(entity, recipients)
+    inlined_jobs do
+      recipients.each do |recipient|
+        DiasporaFederation.callbacks.trigger(:receive_entity, entity, recipient.id)
+      end
+    end
+  end
+
+  def find_private_message(guid)
+    StatusMessage.find_by(guid: guid).tap do |status_msg|
+      expect(status_msg).not_to be_nil
+      expect(status_msg.public?).to be false
+    end
+  end
+
+  def post_status_message_via_federation(text, *recipients)
+    entity = FactoryGirl.build(
+      :status_message_entity,
+      author: remote_raphael.diaspora_handle,
+      text:   text,
+      public: false
+    )
+
+    expect {
+      receive_each(entity, recipients)
+    }.to change(Post, :count).by(1).and change(ShareVisibility, :count).by(recipients.count)
+
+    find_private_message(entity.guid)
+  end
+end
 
 describe "mentioning", type: :request do
   include MentioningSpecHelpers
 
+  RSpec::Matchers.define :be_mentioned_in do |object|
+    include Rails.application.routes.url_helpers
+
+    def do_check_subscribers(object) # only check subscribers for objects legitimate to be sent
+      if object.is_a?(Diaspora::Relayable)
+        object.parent.author.local?
+      else # post
+        object.author.local?
+      end
+    end
+
+    match do |user|
+      object.message.markdownified.include?(person_path(id: user.person.guid)) &&
+        !notifications_about_mentioning(user, object).empty? &&
+        (!do_check_subscribers(object) || object.subscribers.include?(user.person))
+    end
+
+    match_when_negated do |user|
+      notifications_about_mentioning(user, object).empty?
+    end
+  end
+
+  RSpec::Matchers.define :be_in_streams_of do |user|
+    match do |status_message|
+      stream_for(user).map(&:id).include?(status_message.id) &&
+        mention_stream_for(user).map(&:id).include?(status_message.id)
+    end
+
+    match_when_negated do |status_message|
+      !stream_for(user).map(&:id).include?(status_message.id) &&
+        !mention_stream_for(user).map(&:id).include?(status_message.id)
+    end
+  end
+
   before do
-    @user1 = FactoryGirl.create :user_with_aspect
-    @user2 = FactoryGirl.create :user
-    @user3 = FactoryGirl.create :user
+    @user2 = bob
+    @user3 = eve
 
-    @user1.share_with(@user2.person, default_aspect)
-    sign_in @user1
+    expect(users_connected?(alice, bob)).to be true
```

Comment:
What do you think of removing these checks at all and relying on the
fixture builder generated data?